--- pcl-pcl-1.7.2/registration/include/pcl/registration/correspondence_estimation_backprojection.h~	2014-09-10 23:22:57.000000000 +0200
+++ pcl-pcl-1.7.2/registration/include/pcl/registration/correspondence_estimation_backprojection.h	2017-01-26 23:11:14.970602304 +0100
@@ -184,7 +184,7 @@
           * cloud for computing correspondences. By default we use k = 10 nearest 
           * neighbors.
           */
-        inline void
+        inline unsigned int
         getKSearch () const { return (k_); }
         
         /** \brief Clone and cast to CorrespondenceEstimationBase */
--- pcl-pcl-1.7.2/registration/include/pcl/registration/correspondence_estimation_normal_shooting.h~	2014-09-10 23:22:57.000000000 +0200
+++ pcl-pcl-1.7.2/registration/include/pcl/registration/correspondence_estimation_normal_shooting.h	2017-01-26 23:12:02.468654319 +0100
@@ -180,7 +180,7 @@
           * cloud for computing correspondences. By default we use k = 10 nearest 
           * neighbors.
           */
-        inline void
+        inline unsigned int
         getKSearch () const { return (k_); }
 
         /** \brief Clone and cast to CorrespondenceEstimationBase */
