diff -urN pcl-pcl-1.7.2.org/cmake/Modules/FindFZAPI.cmake pcl-pcl-1.7.2/cmake/Modules/FindFZAPI.cmake
--- pcl-pcl-1.7.2.org/cmake/Modules/FindFZAPI.cmake	2014-09-10 23:22:57.000000000 +0200
+++ pcl-pcl-1.7.2/cmake/Modules/FindFZAPI.cmake	2014-11-07 15:22:34.406580761 +0100
@@ -13,7 +13,7 @@
     DOC "Fotonic include directories")
 	
   # Find libraries
-  find_library(FZAPI_LIBS fotonic_fz_api
+  find_library(FZAPI_LIBS fz_api
     HINTS "${FZ_API_DIR}/Release" NO_DEFAULT_PATH
     DOC "Fotonic libraries")
 else()
@@ -23,4 +23,4 @@
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(FZAPI DEFAULT_MSG
                                   FZAPI_LIBS FZAPI_INCLUDE_DIR)
-								  
\ Brak znaku nowej linii na końcu pliku
+								  
